compiler/rustc_session/src/errors.rs

```diff
@@ -375,44 +380,45 @@ pub fn report_lit_error(sess: &ParseSess, err: LitError, lit: token::Lit, span:
     let token::Lit { kind, symbol, suffix, .. } = lit;
     let dcx = &sess.dcx;
     match err {
-        // `LexerError` is an error, but it was already reported
-        // by lexer, so here we don't report it the second time.
-        LitError::LexerError => {}
+        LitError::LexerError => {
+            // Supposedly already reported, but we need an `ErrorGuaranteed`, hence the delayed bug.
+            dcx.span_delayed_bug(span, "lexer error")
```

Comment:
```suggestion
            dcx.span_delayed_bug(span, "lexer error encountered but no error was emitted")
```

or something like that (more descriptive and to make the comment redundant).